### DISCUSSION TITLE
Issue compiling firmware for ender 3 v2 with NO_WORKSPACE_OFFSETS disabled for micro swiss upgrade

### DISCUSSION THREAD

**Discussion by kapab** [mriscoc/Ender3V2S1#236] [Q&A] [answered]
Ender 3 v2 CRtouch Micro swiss

I am trying to set the x y home offset using the M206 gcode but it is showing as an unknown command. I found that NO_WORKSPACE_OFFSETS needs to be disable. I am trying to use the source code to compile the firmware with this disabled. It look like the source code is set up for the SKR board. I made the following changes to what I think the the correct setting for the ender 3 v2 4.2.2 board.
STM32F103RE_creality
BOARD_CREALITY_V422
//#define NO_WORKSPACE_OFFSETS
#define SERIAL_PORT 1
//#define USE_CONTROLLER_FAN 
#define X_DRIVER_TYPE  TMC2208_STANDALONE
#define Y_DRIVER_TYPE  TMC2208_STANDALONE
#define Z_DRIVER_TYPE  TMC2208_STANDALONE
#define E0_DRIVER_TYPE TMC2208_STANDALONE
#define BLTOUCH
#define Z_SAFE_HOMING

I now get this error and can not figure it out. I must not be googling the correct thing. 

Compiling .pio/build/STM32F103RC_creality/src/src/lcd/e3v2/proui/lockscreen.cpp.o
Marlin/src/lcd/e3v2/proui/dwin.cpp: In function 'void SetZOffset()':
Marlin/src/lcd/e3v2/proui/dwin.cpp:2067:7: error: 'babystep' was not declared in this scope
 2067 |       babystep.accum = round(planner.settings.axis_steps_per_mm[Z_AXIS] * BABY_Z_VAR);
      |       ^~~~~~~~
*** [.pio/build/STM32F103RC_creality/src/src/lcd/e3v2/proui/dwin.cpp.o] Error 1
Compiling .pio/build/STM32F103RC_creality/src/src/lcd/e3v2/proui/menus.cpp.o

**Comment by mriscoc** [comment, marked as answer]
You can use proper configuration files from: https://github.com/mriscoc/Special_Configurations for cartesian printers you don't need to enable WORKSPACE OFFSETS. Use min/max X/Y position as it is indicated here: https://github.com/mriscoc/Ender3V2S1/wiki/How-to-set-the-home-offsets

**Reply by kapab** [reply]
The config files are what I needed. Thank you for the help, you rock!
